app/src/main/java/net/gsantner/markor/util/BackupRestoreHelper.java

```diff
@@ -0,0 +1,236 @@
+package net.gsantner.markor.util;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.support.v4.app.FragmentManager;
+import android.text.TextUtils;
+import android.util.Log;
+import android.widget.Toast;
+
+import net.gsantner.markor.R;
+import net.gsantner.markor.format.general.DatetimeFormatDialog;
+import net.gsantner.markor.ui.FilesystemViewerCreator;
+import net.gsantner.markor.ui.SearchReplaceDialog;
+import net.gsantner.markor.ui.hleditor.TextActions;
+import net.gsantner.opoc.ui.FilesystemViewerData;
+import net.gsantner.opoc.util.FileUtils;
+import net.gsantner.opoc.util.GashMap;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class BackupRestoreHelper {
+    private static final String BACKUP_METADATA = "__BACKUP_METADATA__";
+
+    private static final String[] PREF_NAMES = {
+            null, // Default pref
+            AppSettings.SHARED_PREF_APP,
+            DatetimeFormatDialog.DATETIME_SETTINGS,
+            TextActions.ACTION_ORDER_PREF_NAME,
+            SearchReplaceDialog.SEARCH_REPLACE_SETTINGS
+    };
+
+    private static final Pattern[] PREF_EXCLUDE_PATTERNS = {
+            Pattern.compile(".*password.*", Pattern.MULTILINE | Pattern.CASE_INSENSITIVE),
```

Comment:
@gsantner we can't leave this so generic because it will block any per file settings for any file with 'password' in the name.

This affects me, and others too,  I'm sure

Reply:
And what happens then? File settings are taken then for such files with global default values.

Reply:
And what happens then? File settings are taken then for such files with global default values.

On top of that, it will not expose the filepath to your plaintext passwords file you have still laying around somewhere. (Everybody has a passwords txt somewhere, even if its just for test accounts)

Reply:
I am of the opinion that we shouldn't include such hidden special cases or presume the user's intent.

How about if we add a negative look-behind to allow keys which start with `PREF_PREFIX_`? That will allow per file settings while still catching all other possible password related keys.

Reply:
i'd prefer everything with password, but if you say OK with pref_ prefix...also OK